src/Pkg.jl

```diff
@@ -39,31 +28,303 @@ include("Operations.jl")
+A package is tested by running its `test/runtests.jl` file.
+
+The tests are run by generating a temporary environment with only `pkg` and its (recursive) dependencies
+(recursively) in it. If a manifest exist, the versions in that manifest is used, otherwise
+a feasible set of package are resolved and installed.
+
+During the test, test-specific dependencies are active, which are
+given in the project file as e.g.
+
+```
+[extras]
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+
+[targets]
+test = [Test]
+```
+
+Coverage statistics for the packages may be generated by
+passing `coverage=true`. The default behavior is not to run coverage.
+
+The tests are executed in a new process with `check-bounds=yes` and by default `startup-file=no`.
+If using the startup file (`~/.julia/config/startup.jl`) is desired, start julia with `--startup-file=yes`.
+"""
+const test = API.test
+
+"""
+    Pkg.gc()
+
+Garbage collect packages that are no longer reachable from any project.
+Only packages that are tracked by version are deleted, so no packages
+that might contain local changes are touched.
+"""
+const gc = API.gc
+
+
+"""
+    Pkg.build()
+    Pkg.build(pkg::Union{String, Vector{String})
+    Pkg.build(pkgs::Union{PackageSpec, Vector{PackageSpec}})
+
+Run the build script in `deps/build.jl` for `pkg` and all of the dependencies in
+depth-first recursive order.
+If no argument is given to `build`, the current project is built, which thus needs
+to be a package.
+This function is called automatically one any package that gets installed
+for the first time.
+"""
+const build = API.build
+
+# TODO: decide what to do with this
+const installed = API.installed
+
+"""
+    Pkg.pin(pkg::Union{String, Vector{String})
+    Pkg.pin(pkgs::Union{Packagespec, Vector{Packagespec}})
+
+Pin a package to the current version (or the one given in the `packagespec` or a certain
+git revision. A pinned package is never updated.
+"""
+const pin = API.pin
+
+"""
+    Pkg.free(pkg::Union{String, Vector{String})
+    Pkg.free(pkgs::Union{Packagespec, Vector{Packagespec}})
+
+Free a package which removes a `pin` if it exists, or if the package is tracking a path,
+e.g. after [`Pkg.develop`](@ref), go back to tracking registered versions.
+
+# Examples
+```julia
+Pkg.free("Package")
+Pkg.free(PackageSpec("Package"))
+```
+"""
+const free = API.free
+
+
+"""
+    Pkg.develop(pkg::Union{String, Vector{String})
+    Pkg.develop(pkgs::Union{Packagespec, Vector{Packagespec}})
+
+Make a package available for development by tracking it by path.
+If `pkg` is given with only a name or by a URL the packages will be downloaded
+to the location by the environment variable `JULIA_PKG_DEVDIR` with
+`.julia/dev` as the default.
+
+If `pkg` is given as a local path, the package at that path will be tracked.
+
+# Examples
+```julia
+# By name
+Pkg.develop("Example")
+
+# By url
+Pkg.develop(PackageSpec(url="https://github.com/JuliaLang/Compat.jl", rev="master"))
```

Comment:
@KristofferC, has this been solved in master?
```
julia> Pkg.develop(PackageSpec(url="https://github.com/JuliaLang/Compat.jl", rev="master"))
ERROR: git revision cannot be given to `develop`
Stacktrace:
 [1] pkgerror(::String) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Types.jl:120
 [2] #add_or_develop#13(::Symbol, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:39
 [3] #add_or_develop at ./none:0 [inlined]
 [4] #add_or_develop#12 at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:29 [inlined]
 [5] #add_or_develop at ./none:0 [inlined]
 [6] #add_or_develop#10(::Base.Iterators.Pairs{Symbol,Any,Tuple{Symbol,Symbol},NamedTuple{(:mode, :shared),Tuple{Symbol,Bool}}}, ::Function, ::Pkg.Types.PackageSpec) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:27
 [7] #add_or_develop at ./none:0 [inlined]
 [8] #develop#19 at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:70 [inlined]
 [9] develop(::Pkg.Types.PackageSpec) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:70
 [10] top-level scope at none:0
```

Reply:
It's not supposed to work, and it is not documented anymore.

Reply:
Should people just checkout the branch through Git/vcs after `dev` gets the repo?

Reply:
Yes.